version: '3.7'

x-django-base:
  &django-base
  build: .
  volumes:
    - "./data/static:/static"
    - "./data/media:/website/media"
  restart: unless-stopped
  environment:
    - DJANGO_SETTINGS_MODULE=website.settings
    - WEBSITE_DEBUG=False
    - WEBSITE_CELERY_BROKER_INLINE=true
    - C_FORCE_ROOT=true

services:
  db:
    image: postgres:14.2
    volumes:
      - ./data/db/14/data:/var/lib/postgresql/data
    restart: unless-stopped
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres

  redis:
    image: redis:6
    restart: unless-stopped

  nginx:
    restart: unless-stopped
    image: nginx
    volumes:
      # - ./conf/uwsgi_params:/conf/uwsgi_params:ro
    - ./conf/nginx.conf:/etc/nginx/nginx.conf
    - ./data/static:/usr/share/nginx/html/static:ro
    - ./data/media:/usr/share/nginx/html/media:ro
    depends_on:
      - django
      - db
    networks:
      - proxy
      - default
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.website-server.rule=Host(`30er.party`)"
      - "traefik.http.routers.website-server.middlewares=redirect-to-https,test-auth"
      - "traefik.http.services.website-server.loadbalancer.server.port=8000"
      - "traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https"
      - "traefik.http.middlewares.test-auth.basicauth.users=stefan:$apr1$lKBFGi9O$ND39El3VXJZJ7IRufPaJs0"

  django:
    <<: *django-base
    networks:
      - default
      - proxy
    command: /entrypoint.sh

  celery-worker:
    <<: *django-base
    command: poetry run python -m celery -A website.celery_conf worker --loglevel=INFO --concurrency=1
    deploy:
      resources:
        limits:
          cpus: '1.0'
          memory: 3G


networks:
  proxy:
    external: true
